x-pack/plugins/observability_solution/observability_onboarding/public/assets/auto_detect.sh

```diff
@@ -0,0 +1,177 @@
+
+  known_integrations_list_string=$(echo -e "$known_integrations_list_string" | sort -u)
+}
+
+known_integration_title() {
+  local integration=$1
+  case $integration in
+    "nginx")
+      echo "Nginx Logs"
+      ;;
+    "apache")
+      echo "Apache Logs"
+      ;;
+    "docker")
+      echo "Docker Container Logs"
+      ;;
+    "system")
+      echo "System Logs"
+      ;;
+    *)
+      echo "Unknown"
+      ;;
+  esac
+}
+
+build_unknown_log_file_patterns() {
+  while IFS= read -r log_file_path; do
+    unknown_log_file_pattern_list_string+="$(dirname "$log_file_path")/*.log\n"
+  done <<< "$(echo -e $unknown_log_file_path_list_string)"
+
+  unknown_log_file_pattern_list_string=$(echo -e "$unknown_log_file_pattern_list_string" | sort -u)
+}
+
+function select_list() {
+  local known_integrations_options=()
+  local unknown_logs_options=()
+
+  while IFS= read -r line; do
+    if [[ -z "$line" ]]; then
+      continue
+    fi
+    known_integrations_options+=("$line")
+  done <<< "$known_integrations_list_string"
+
+  while IFS= read -r line; do
+    if [[ -z "$line" ]]; then
+      continue
+    fi
+    unknown_logs_options+=("$line")
+  done <<< "$unknown_log_file_pattern_list_string"
+
+  local options=("${known_integrations_options[@]}" "${unknown_logs_options[@]}")
+
+  for i in "${!options[@]}"; do
+    if [[ "$i" -lt "${#known_integrations_options[@]}" ]]; then
+      printf "\e[32m%s)\e[0m %s\n" "${i}" "$(known_integration_title "${options[$i]}")"
+    else
+      printf "\e[32m%s)\e[0m %s\n" "${i}" "${options[$i]}"
+    fi
+  done
+
+  echo -e "\n"
+  read -p "Ingest all of them? [Y/n] (default: Yes): " confirmation_reply
+  confirmation_reply="${confirmation_reply:-Y}"
+
+  if [[ ! "$confirmation_reply" =~ ^[Yy]$ ]]; then
+    echo -e "\nExclude logs by listing their index numbers (e.g. 1, 2, 3):"
+    read exclude_index_list_string
+
+    IFS=', ' read -r -a exclude_index_list_array <<< "$exclude_index_list_string"
+
+    for index in "${!options[@]}"; do
+      local is_excluded=0
+      for excluded_index in "${exclude_index_list_array[@]}"; do
+        if [[ "$index" -eq "$excluded_index" ]]; then
+            is_excluded=1
+        fi
+      done
+
+      if [[ $is_excluded -eq 0 ]]; then
+        if [[ "$index" -lt "${#known_integrations_options[@]}" ]]; then
+          selected_known_integrations_array+=("${options[index]}")
+        else
+          selected_unknown_log_file_pattern_array+=("${options[index]}")
+        fi
+      fi
+    done
+  else
+    selected_known_integrations_array=("${known_integrations_options[@]}")
+    selected_unknown_log_file_pattern_array=("${unknown_logs_options[@]}")
+  fi
+}
+
+echo "Looking for log files..."
+read_log_file_list
+detect_known_integrations
+build_unknown_log_file_patterns
+
+echo -e "\nWe found these logs on your system:"
+select_list
+
+echo -e "\nList any custom logs that we have not detected (e.g. /var/log/myapp/*.log, /home/j/myapp/*.log). Press Enter to skip."
```

Comment:
I think one example is enough here. The prompt is quite long as is

Reply:
Yeah, I wanted to highlight somehow that you can enter multiple patterns, maybe Christian will have some thoughts about the copy here, I'll ask him.

Reply:
Oh I see, yeh that's a good point. Let's keep it then!